### PR TITLE
chore(logging): use structlog for stdlib log and gunicorn

### PR DESCRIPTION
**Pull request by hazzadous**
Change ensures that we have a common format for all log lines,
additionally adding e.g. pid and tid for gunicorn logs.

Previously there were large plain text log lines over multiple lines
that results in very awkward to logs when ingested into e.g. CloudWatch
or Loki, not least because these log apps typically show the most recent
line first.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: gunicorn.config.py [hunk 1]
```diff
@@ -187,3 +166,78 @@ def run(self):
...
+    # Add the log level and a timestamp to the event_dict if the log entry
+    # is not from structlog.
+    structlog.stdlib.add_log_level,
+    structlog.stdlib.add_logger_name,
+    add_pid_and_tid,
+    structlog.processors.TimeStamper(fmt="iso"),
+]
+
+
+# This is a copy the default logging config for gunicorn but wtith additions to:
```
Comment (guidoiaquinti) [inline gunicorn.config.py:192]:
```suggestion
# This is a copy the default logging config for gunicorn but with additions to:
```

**Review by ellie — APPROVED** [review]
nice to kill some print statements 🙌

File: posthog/utils.py [hunk 2]
```diff
@@ -899,12 +900,8 @@ def str_to_bool(value: Any) -> bool:
     return str(value).lower() in ("y", "yes", "t", "true", "on", "1")
 
 
-def print_warning(warning_lines: Sequence[str], *, top_emoji="🔻", bottom_emoji="🔺"):
-    highlight_length = min(max(map(len, warning_lines)) // 2, shutil.get_terminal_size().columns)
-    print(
-        "\n".join(("", top_emoji * highlight_length, *warning_lines, bottom_emoji * highlight_length, "",)),
-        file=sys.stderr,
-    )
+def print_warning(warning_lines: List[str]):
```
Comment (guidoiaquinti) [inline posthog/utils.py:903]: Follow up PR, replace all of those with direct calls to `logger.warning`

**Review by guidoiaquinti — APPROVED** [review]
LGTM but mypy is failing

File: posthog/utils.py (hunk 2 above)
Comment (hazzadous) [reply posthog/utils.py:903]: I might just do that now, I didn't want to change too much in one go but I think it makes sense.

**Pull request merged**